addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/CorsScanRuleUnitTest.java

```diff
@@ -0,0 +1,149 @@
+
+import fi.iki.elonen.NanoHTTPD.IHTTPSession;
+import fi.iki.elonen.NanoHTTPD.Response;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+
+/** Unit test for {@link CorsScanRule}. */
+public class CorsScanRuleUnitTest extends ActiveScannerTest<CorsScanRule> {
+    private static final String ACAC = "Access-Control-Allow-Credentials";
+    private static final String GENERIC_RESPONSE =
+            "<!DOCTYPE HTML PUBLIC \"-//IETF//DTD HTML 2.0//EN\">\n"
+                    + "<html><head></head><body></body></html>";
+
+    @Override
+    protected CorsScanRule createScanner() {
+        return new CorsScanRule();
+    }
+
+    @Test
+    public void shouldNotAlertIfCorsNotSupported() throws Exception {
+        // Given
+        nano.addHandler(new CORSResponse(null, false));
+        HttpMessage msg = this.getHttpMessage("/");
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+    }
+
+    @Test
+    public void shouldAlertInfoIfACAOButNotPayloads() throws Exception {
+        // Given
+        nano.addHandler(new CORSResponse("dummyValue", false));
+        HttpMessage msg = this.getHttpMessage("/");
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertRisk(Alert.RISK_INFO);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"REFLECT", "*", "null"})
+    public void shouldAlertMediumIfACAOPayloads(String origin) throws Exception {
+        // Given
+        nano.addHandler(new CORSResponse(origin, false));
+        HttpMessage msg = this.getHttpMessage("/");
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertRisk(Alert.RISK_MEDIUM);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"REFLECT", "null"})
+    public void shouldAlertHighIfACAOAndACACPayloads(String origin) throws Exception {
+        // Given
+        nano.addHandler(new CORSResponse(origin, true));
+        HttpMessage msg = this.getHttpMessage("/");
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertRisk(Alert.RISK_HIGH);
+    }
+
+    @Test
+    public void shouldAlertMediumIfACAOWildcardAndACAC() throws Exception {
+        // Given
+        nano.addHandler(new CORSResponse("*", true));
+        HttpMessage msg = this.getHttpMessage("/");
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertRisk(Alert.RISK_MEDIUM);
+    }
+
+    private void assertRisk(int risk) {
```

Comment:
Maybe rename this to assertExpectedAlert or something like that since it checks # raised as well as risk.